data/data/gcp/cluster/master/main.tf

```diff
@@ -61,6 +61,19 @@ resource "google_compute_instance" "master" {
     kms_key_self_link = var.root_volume_kms_key_link
   }
 
+  dynamic "confidential_instance_config" {
+    for_each = var.enable_confidential_compute ? [1] : []
+    content {
+      enable_confidential_compute = true
+    }
+  }
```

Comment:
I'm not sure I'm following this syntax, but it seems overly complex. I suspect we don't need a dynamic block here at all. Can we simplify this to just:

```
  name         = "${var.cluster_id}-master-${count.index}"
  machine_type = var.machine_type
  zone         = element(var.zones, count.index)
  enable_confidential_compute = var.enable_confidential_compute
  on_host_maintenance = var.on_host_maintenance
```

Reply:
Hmm, I added the dynamic block because the TF should look like this:
```
confidential_instance_config {
   enable_confidential_compute = var.enable_confidential_compute
}

scheduling {
   on_host_maintenance = var.on_host_maintenance
}
```
Also, I'd like to add these blocks only in case the user specified the configuration in the install config, otherwise, I'd like to  let the GCP platform choose the default